akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -931,12 +931,24 @@ private[akka] class ShardRegion(
         val coordinatorMessage =
           if (cluster.state.unreachable(membersByAge.head)) s"Coordinator [${membersByAge.head}] is unreachable."
           else s"Coordinator [${membersByAge.head}] is reachable."
-        log.warning(
-          "{}: Trying to register to coordinator at [{}], but no acknowledgement. Total [{}] buffered messages. [{}]",
-          typeName,
-          actorSelections.mkString(", "),
-          shardBuffers.totalSize,
-          coordinatorMessage)
+        if (shardBuffers.totalSize > 0) {
+          if (log.isWarningEnabled) {
```

Comment:
added this due to the mkString

Reply:
hmm, we need the if to choose between ward and debug. The `mkString` will happen anyway as it is now. 

Unless we guard the debug block as well.